### PR TITLE
Code Quality: Improve the implementation of StatusCenterItemProgressModel

### PR DESCRIPTION
**Pull request by hez2010**
**Resolved / Related Issues**
   Contributes to #11265

This PR provides the fundamental support needed by transfer speed implementation. 
/cc: @0x5bfa 

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [x] Did you check for accessibility? You can use Accessibility Insights for this.

### CODE REVIEW

File: src/Files.App/Utils/StatusCenter/StatusCenterItemProgressModel.cs [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
+using System.Collections.Concurrent;
+using System.Runtime.CompilerServices;
+
 namespace Files.App.Utils.StatusCenter
 {
 	/// <summary>
 	/// Represents a model for file system operation progress.
 	/// </summary>
-	public class StatusCenterItemProgressModel : ObservableObject
+	public class StatusCenterItemProgressModel : INotifyPropertyChanged
```
Comment (0x5bfa) [inline src/Files.App/Utils/StatusCenter/StatusCenterItemProgressModel.cs:12]: Why did you changed from ObservableObject to INotifyPropertyChanged?

**Review by 0x5bfa — COMMENTED** [review]
lgtm

File: src/Files.App/Utils/StatusCenter/StatusCenterItemProgressModel.cs (hunk 1 above)
Comment (hez2010) [reply src/Files.App/Utils/StatusCenter/StatusCenterItemProgressModel.cs:12]: See the below `SetProperty` method.

I need to use a dirty tracker and then bulk trigger the update with a debouncer, so we cannot use the `ObservableObject` which will notify the update on every change happening.

Comment (0x5bfa) [reply src/Files.App/Utils/StatusCenter/StatusCenterItemProgressModel.cs:12]: Thank you for explanation!

**Review by 0x5bfa — APPROVED** [review]
LGTM. Extreme thanks to you!

**Pull request merged**